### ISSUE TITLE
PrimeNG@4.2.0-rc.1 -> DataTable constructor threw runtime exception 

### ISSUE THREAD

**Issue by greengalaxy2016** [primefaces/primeng#3822]
### There is no guarantee in receiving a response in GitHub Issue Tracker, If you'd like to secure our response, you may consider *PrimeNG PRO Support* where support is provided within 4 business hours

**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please fork the plunkr below and create a case demonstrating your bug report. Issues without a plunkr have much less possibility to be reviewed.

http://plnkr.co/edit/Qi2Nw2imLpDUF8NJinYw?p=preview

**Current behavior**
PrimeNG version was 1.1.4. My application uses \<p-dataTable\> in the template and run well, but after upgraded to PrimeNG 4.2.0-rc.1, (also tried 4.1.3 and same exception was thrown) the following runtime exception was thrown: 
`
app-187a9b0723.js:34 Error: Uncaught (in promise): Error: Can't resolve all parameters for DataTable: (e, DomHandler, e, ?, e, ObjectUtils, e).
Error: Can't resolve all parameters for DataTable: (e, DomHandler, e, ?, e, ObjectUtils, e).
    at m (lib-a9b65af493.js:3)
    at e.getDependenciesMetadata (app-187a9b0723.js:9)
    at e.getTypeMetadata (app-187a9b0723.js:9)
    at e.getDirectiveMetadata (app-187a9b0723.js:9)
    at app-187a9b0723.js:9
    at Array.forEach (<anonymous>)
    at e.getNgModuleMetadata (app-187a9b0723.js:9)
    at app-187a9b0723.js:9
    at Array.forEach (<anonymous>)
    at e.getNgModuleMetadata (app-187a9b0723.js:9)
    at app-187a9b0723.js:9
    at Array.forEach (<anonymous>)
    at e.getNgModuleMetadata (app-187a9b0723.js:9)
    at e._compileComponents (app-187a9b0723.js:10)
    at e._compileModuleAndComponents (app-187a9b0723.js:10)
    at m (lib-a9b65af493.js:3)
    at l (lib-a9b65af493.js:3)
    at lib-a9b65af493.js:3
    at lib-a9b65af493.js:4
    at t.invoke (lib-a9b65af493.js:3)
    at Object.onInvoke (app-187a9b0723.js:35)
    at t.invoke (lib-a9b65af493.js:3)
    at n.run (lib-a9b65af493.js:3)
    at lib-a9b65af493.js:3
    at t.invokeTask (lib-a9b65af493.js:4)
    at Object.onInvokeTask (app-187a9b0723.js:35)
    at t.invokeTask (lib-a9b65af493.js:4)
    at n.runTask (lib-a9b65af493.js:3)
    at a (lib-a9b65af493.js:3)
    at XMLHttpRequest.invoke (lib-a9b65af493.js:4)
`

**Expected behavior**
application which uses DataTable should not threw exception at runtime. 

**Minimal reproduction of the problem with instructions**

**What is the motivation / use case for changing the behavior?**
a major blocker

**Please tell us about your environment:**
RedHat 7, IntelJ, npm -version = 3.10.10 

* **Angular version:** 4.0.0

* **PrimeNG version:** 4.2.0-rc.1

* **Browser:** [all | Chrome XX | Firefox XX ]
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
@types/node: ^8.0.26

* **Node (for AoT issues):** `node --version` = 6.10.2

**Comment by greengalaxy2016** [comment]
`
<p-dataTable id="dtTransactions" [tableStyleClass]="full-table" [totalRecords]="totalRecords" [lazy]="true" (onPage)="lazyLoadData($event)"
                       [value]="searchResults" [rows]="3" [paginator]="true" [rowsPerPageOptions]="[3,6,9]" [pageLinks]="3"  [emptyMessage]="''">
            <p-column field="brand" header="Brand" [sortable]="true" [style]="{'width': '8%'}"></p-column>
            <p-column field="createdDate" header="Created" [sortable]="true" [style]="{'width': '10%'}"></p-column>
            <p-column field="transactionId" header="Transaction ID" [sortable]="true" [style]="{'width': '10%'}">
              <template let-col let-transaction="rowData" pTemplate="body">
                <a (click)="onRowSelect($event,transaction)">{{transaction[col.field]}}</a>
              </template>
            </p-column>
            <p-column field="paymentRequestId" header="Group ID" [sortable]="true" [style]="{'width': '8%'}"></p-column>
            <p-column field="status" header="Status" [sortable]="true" [style]="{'width': '8%'}"></p-column>
            <p-column field="type" header="Type(Payment Request)" [sortable]="true" [style]="{'width': '6%'}"></p-column>
            <p-column field="product" header="Product" [sortable]="true" [style]="{'width': '8%'}"></p-column>
            <p-column field="amount" header="Amount" [sortable]="true" [style]="{'width': '6%'}"></p-column>
            <p-column field="feeAmount" header="Fee" [sortable]="true" [style]="{'width': '8%'}"></p-column>
            <p-column field="currency" header="Currency" [style]="{'width': '6%'}"></p-column>
            <p-column field="ipCountry" header="IP Country" [sortable]="true" [style]="{'width': '6%'}"></p-column>
          </p-dataTable>
`

**Comment by cagataycivici** [comment]
@Merve7 Review please.

**Comment by greengalaxy2016** [comment]
I tried to reproduce in plunkr, but it works in plunkr. so I checked the version of primeng and angular in plunk system.config.js which is 4.1.0 and 4.2.6 respectively. I changed the version in my application to the same version, but still get the exception at runtime. Here is my package.json and system.config.js (p.s. all these works with primeng@1.1.4) 

[package-system-configjs.zip](https://github.com/primefaces/primeng/files/1281734/package-system-configjs.zip)
